code/modules/mob/living/simple_animal/hostile/mining/goliath.dm

```diff
@@ -195,3 +470,96 @@
 	icon_state = "Goliath_tentacle_retract"
 	deltimer(timerid)
 	timerid = QDEL_IN(src, 7)
+
+/mob/living/simple_animal/hostile/asteroid/goliath/proc/add_draconian_effect() // Apply the draconian overlay
+	cut_overlay(draconian_overlay)
+	var/status
+	if(draconian == NOT_DRACONIAN)
+		return
+	if(stat == DEAD)
+		status = "_dead"
+	if(growth_stage == JUVENILE)
+		draconian_overlay = mutable_appearance('icons/mob/lavaland/lavaland_monsters.dmi', "goliath_glow_juvenile[status]")
+	else if(growth_stage == SUBADULT)
+		draconian_overlay = mutable_appearance('icons/mob/lavaland/lavaland_monsters.dmi', "goliath_glow_subadult[status]")
+	else if(growth_stage == ADULT)
+		draconian_overlay = mutable_appearance('icons/mob/lavaland/lavaland_monsters.dmi', "goliath_glow_adult[status]")
+	if(draconian == DRACONIAN)
+		draconian_overlay.alpha = 150
+	if(draconian == FULL_DRACONIAN)
+		draconian_overlay.alpha = 255
+	add_overlay(draconian_overlay)
+
+/mob/living/simple_animal/hostile/asteroid/goliath/attackby(obj/item/O, mob/user, params) // Feeding to tame
+	if(istype(O, food_wanted))
+		if(growth_stage == ADULT && growth_stage == ANCIENT)
+			return
+		if(!stat && feed_cooldown <= 0)
+			switch(food_wanted)
+				if(MEAT)
+					tame_progress += rand(200, 350)
+				if(DIAMOND)
+					var/obj/item/stack/ore/diamond/D = O
+					if(D.amount != 1)
+						to_chat(user, "<span class='warning'>\The [src] only wants one diamond!</span>") // So that you don't accidentally feed more than one diamond
+						return
+					else
+						tame_progress += rand(250, 400)
+				if(CORE)
+					var/obj/item/organ/internal/regenerative_core/R = O
+					if(!R.inert)
+						tame_progress += rand(250, 450)
+					else
+						tame_progress += rand(150, 250) // Inert cores don't count a lot
+						visible_message("<span class='notice'>\The [src] didn't like [O] too much...</span>")
+				if(FLORA)
+					tame_progress += rand(150, 350)
+				if(ORGANS)
+					if((!istype(O, CORE)) || (!istype(O, CYBERORGAN))) // No legion or cyberimp organs if it wants organs.
+						tame_progress += rand(300, 500)
+			user.visible_message("<span class='notice'>[user] feeds [O] to [src].</span>")
+			playsound(get_turf(src), 'sound/items/eatfood.ogg', 50, 0)
+			user.drop_item()
+			qdel(O)
+			if(food_wanted == MEAT)
+				feed_cooldown = rand(5, 10)
+			else
+				feed_cooldown = rand(30, 100)
+			reroll_food()
+		else // If dead or not hungry
+			if(stat == DEAD)
+				to_chat(user, "<span class='warning'>\The [src] is dead!</span>")
+			else if(feed_cooldown >= 1)
+				to_chat(user, "<span class='warning'>\The [src] is not hungry yet!</span>")
+	else if(istype(O, DRAGONSBLOOD) && draconian <= 1)
+		if(!stat)
+			user.visible_message("<span class='notice'>[user] feeds [O] to [src].</span>")
+			user.drop_item()
+			qdel(O)
+			playsound(get_turf(src), 'sound/items/drink.ogg', 50, 0)
+			tame_progress += rand(950, 1200)
+			maxHealth = maxHealth += 100
```

Comment:
Double assignment here. Same as the one below.
`maxHealth += 100` compiles to: `maxHealth = maxHealth + 100`